Makefile

```diff
@@ -1,14 +1,23 @@
+CONTAINER_TOOL := docker
+
+podman: CONTAINER_TOOL = podman
+podman: build-test check-podman
+
 all: build-test check
 
 build-test: clean
-	@docker build -t diff-shellcheck-test-image test/
+	$(CONTAINER_TOOL) build -t diff-shellcheck-test-image test/
 
-check:
-	@docker run -dt -v "${PWD}:/home" --name diff-shellcheck-test diff-shellcheck-test-image
-	@docker exec diff-shellcheck-test bash -c 'cd /home && bats test'
-	@docker container stop diff-shellcheck-test
+check: check-$(CONTAINER_TOOL)
+check-docker:
+	$(CONTAINER_TOOL) run -t -v "${PWD}:/home" --name diff-shellcheck-test diff-shellcheck-test-image 'cd /home && bats test'
+check-podman:
+	$(CONTAINER_TOOL) run -t -v "${PWD}:/home:O" --name diff-shellcheck-test diff-shellcheck-test-image 'cd /home && bats test'
 
-clean:
-	@docker container rm -f diff-shellcheck-test
+clean: clean-podman clean-docker
+clean-podman:
+	podman container rm -f diff-shellcheck-test
+clean-docker:
+	docker container rm -f diff-shellcheck-test
 
-.PHONY: all build-test check clean
+.PHONY: podman all build-test check clean clean-podman clean-docker
```

Comment:
Fails with:
```console
make podman
podman container rm -f diff-shellcheck-test
docker container rm -f diff-shellcheck-test
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

```suggestion
clean:
 	$(CONTAINER_TOOL) container rm -f diff-shellcheck-test

.PHONY: podman all build-test check clean
```

Reply:
Hmm, that does not work as expected :/

Reply:
Yeah, there is an explanation:
- https://www.unix.com/programming/118420-makefile-ifeq-not-working.html

Reply:
Sorry, the problem is, that even without this change, plain `make` tries to run everything in `podman`, which is not we want, right?

Reply:
Also, the new `check-*` targets are missing from `.PHONY`.